File: app/src/main/java/com/asfoundation/wallet/di/MyAddressModule.java
package com.asfoundation.wallet.di;

import com.asfoundation.wallet.router.TransactionsRouter;
import com.asfoundation.wallet.viewmodel.MyAddressViewModelFactory;
import dagger.Module;
import dagger.Provides;

@Module public class MyAddressModule {

  @Provides MyAddressViewModelFactory providesMyAddressViewModelFactory(
      TransactionsRouter transactionsRouter) {
    return new MyAddressViewModelFactory(transactionsRouter);
  }

  @Provides TransactionsRouter provideTransactionsRouter() {
    return new TransactionsRouter();
  }
}

File: app/src/main/java/com/asfoundation/wallet/di/MyAddressModule.kt
package com.asfoundation.wallet.di

import com.asfoundation.wallet.router.TransactionsRouter
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

@InstallIn(ActivityComponent::class)
@Module
class MyAddressModule {

  @Provides
  fun provideTransactionsRouter() = TransactionsRouter()
}